experimental/PropertyDDS/packages/property-properties/src/index.d.ts

```diff
@@ -769,97 +769,103 @@ declare module "@fluid-experimental/property-properties" {
-             * @throws if tyring to modify a referenced property.
-             */
-            insertRange<T = any>(in_offset: number, in_array: T[]): void;
-            /**
-             * Removes a given number of elements from the array property (or given number of letters from a StringProperty)
-             *  and shifts remaining values to the left.
-             * E.g. [1, 2, 3, 4, 5]  .removeRange(1, 3) => [1, 5]
-             */
-            removeRange(): Array<any> | Array<BaseProperty>;
-            /**
-             * Sets the array properties elements to the content of the given array
-             * All changed elements must already exist. This will overwrite existing elements.
-             * E.g. [1, 2, 3, 4, 5]  .setRange(1, [7, 8]) => [1, 7, 8, 4, 5]
-             */
-            setRange(in_offset: number, in_array: Array<any>): void;
-            /**
-             * Returns an object with all the nested values contained in this property
-             */
-            getValues<T = BaseProperty[]>(): T;
-            getLength(): number;
-            /**
-             * Checks whether a property or data exists at the given position.
-             * @param in_position index of the property
-             */
-            has(in_position: string): boolean;
-            /**
-             * Returns the full property type identifier for the ChangeSet including the array type id, if not
-             * omitted by parameters
-             * @param in_hideCollection - if true the collection type (if applicable) will be omitted. Default to false
-             * @return The typeid
-             */
-            public getFullTypeid(in_hideCollection?: boolean): string
+          /**
+           * Default constructor for ArrayProperty
+           * @param in_params the parameters
+           * @param in_constructor the constructor for the array data
+           * @param in_primitiveType Is this an array of primitive types?
+           * @param in_scope The scope in which the property typeid is defined
+           */
+          constructor(in_params: ArrayProperty_ArrayProperty_in_params_TYPE, in_constructor: object, in_primitiveType: Boolean, in_scope: string | undefined);
+
+          public get length()
+          /**
+           * Gets the array element at a given index
+           * @param in_position the target index
+           *   if an array is passed, elements in the array will be treated as part of a path.
+           *   The first item in an array should be a position in the array.
+           *   For example, .get([0,'position','x']) is the equivalent of .get(0).get('position').get('x')
+           *   If it encounters a ReferenceProperty, .get will, by default, resolve the property it refers to.
+           * @param in_options parameter object
+           */
+          get<T=BaseProperty>(in_position: number | string | Array<string|number>, in_options?: ArrayProperty_get_in_options_TYPE): T | undefined;
```

Comment:
Did you also want to that 'T extend BaseProperty'?
```ts
get<T extends BaseProperty = BaseProperty>(...) : T | undefined
```

Reply:
I'll merge "as-is" and you can bundle the 'extends' clause in your next PR (if it's desired.)